frontend/app/actions/catDetailActions.js

```diff
@@ -0,0 +1,45 @@
+import { RSAA } from 'redux-api-middleware'
+import { forEach } from 'lodash'
+
+export const REQUEST = 'CAT_DETAIL_REQUEST'
+export const SUCCESS = 'CAT_DETAIL_SUCCESS'
+export const FAILURE = 'CAT_DETAIL_FAILURE'
+
+function cleanCatDetail(pet) {
```

Comment:
this seems like a fine solution to me!

looking at the cat detail object in the output, it seems to get you exactly what you were looking for.

Reply:
You can probably re-use this clean function for other endpoint responses if they are similar. That would be better than writing a separate one for each endpoint.

Reply:
actually a couple comments on this if you want to take it further:

1) it's generally a very bad idea in any language to mute a list or object as you are iterating through it. You seem to have gotten lucky in this case, but better practice is to build up a second "clean" object as you are iterating through. Then just return that object from the function.

2) if you wanted to make this generalized, you could use similar logic with a recursion or a stack instead of two nested loops. that might be a rabbit hole though, so you're better off keeping it simple.